crates/synthesis/src/linalg/mod.rs

```diff
@@ -12,15 +12,79 @@
 
 use approx::{abs_diff_eq, relative_ne};
 use faer::MatRef;
-use faer_ext::{IntoFaer, IntoNalgebra};
-use nalgebra::{DMatrix, DMatrixView};
+use faer_ext::IntoFaer;
+use nalgebra::{DMatrix, DMatrixView, Dim, Dyn, MatrixView, ViewStorage};
+use ndarray::ArrayView2;
+use ndarray::ShapeBuilder;
 use num_complex::Complex64;
 
 pub mod cos_sin_decomp;
 
 const ATOL_DEFAULT: f64 = 1e-8;
 const RTOL_DEFAULT: f64 = 1e-5;
 
+fn nalgebra_to_faer<R: Dim, C: Dim, RStride: Dim, CStride: Dim>(
```

Comment:
Should this (and `faer_to_nalgebra`) be more public than this, or are we deliberately limiting scope?